packages/components/src/SidePanel/SidePanel.component.js

```diff
@@ -91,73 +93,75 @@ function SidePanel({
 	const expandLabel = t('SIDEPANEL_EXPAND', { defaultValue: 'Expand' });
 	const collapseTitle = t('SIDEPANEL_COLLAPSE', { defaultValue: 'Collapse' });
 	const toggleButtonTitle = docked ? expandLabel : collapseTitle;
-
+	const Components = Inject.getAll(getComponent, { Action });
 	return (
 		<nav className={navCSS} role="navigation">
+			{injected('preul')}
 			<ul className={listCSS}>
 				{dockable && (
 					<li className={theme['toggle-btn']} title={toggleButtonTitle}>
-						<Action
+						<Components.Action
 							id={id && `${id}-toggle-dock`}
-							className={theme.link}
 							bsStyle="link"
 							onClick={onToggleDock}
 							icon="talend-opener"
 							label=""
 						/>
 					</li>
 				)}
-				{actions.map(action => {
-					const a11y = {};
-					const extra = {};
-					const isSelected = isActionSelected(action);
+				{injected('beforeactions')}
```

Comment:
beforeActions and actions imply that we know the implementation, to inject li tags.
We should reshape this, by adding a div with toggle button (outside of ul), then ul/li are rendered only if we have props.actions. Else, we render the injected actions, completely independent from ul/li.